### PR TITLE
[Fix][Flink] Resolve streaming source high CPU usage

### LINKED ISSUE #8213: [Bug] [Flink] 使用flink引擎执行Seatunnel cdc任务时，source端一直处于busy状态，这个是极不正常的
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

使用flink引擎执行Seatunnel cdc任务时，source端一直处于busy状态
以下是我的配置文件：
env{
  job.mode="STREAMING"
  parallelism = 1
  job.name= "ft_MysqlToDoris"
  #flink.state.checkpoint-storage="filesystem"
  #flink.state.checkpoints.dir="file:///D:/tmp/seatunnel/linux/"

}
source{
 MySQL-CDC {
	base-url ="jdbc:mysql://localhost:3306/test"
	username = "root"
	password = "123456"
	table-names = ["test.test"]
	table-names-config = [{"table": "test.test", "primaryKeys": ["id"]}]
}

}
transform{

}
sink{

 Console {
   parallelism = 2
 }
}
数据源端只有两条数据，从提交任务开始，算子状态即为busy。

### SeaTunnel Version

2.3.8

### SeaTunnel Config

```conf
env{
  job.mode="STREAMING"
  parallelism = 1
  job.name= "ft_MysqlToDoris"
  #flink.state.checkpoint-storage="filesystem"
  #flink.state.checkpoints.dir="file:///D:/tmp/seatunnel/linux/"

}
source{
 MySQL-CDC {
	base-url ="jdbc:mysql://localhost:3306/test"
	username = "root"
	password = "123456"
	table-names = ["test.test"]
	table-names-config = [{"table": "test.test", "primaryKeys": ["id"]}]
}

}
transform{

}
sink{

 Console {
   parallelism = 2
 }
}
```

### Running Command

```shell
我使用源码编译的方式启动程序
org.apache.seatunnel.example.flink.v2.SeaTunnelApiExample
```

### Error Exception

```log
无明显日志报错，程序执行正常，但source算子长期处于busy 100%状态，并且状态不会更新
```

### Zeta or Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by zhangshenghang**
### Purpose of this pull request

- close #8213 

#### Verify Source  Kafka
image
#### Verify Source  MySQL-CDC
image

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java [hunk 1]
```diff
@@ -55,6 +58,18 @@ public class FlinkSourceReader<SplitT extends SourceSplit>
...
+
+    private static final long DEFAULT_WAIT_TIME_MILLIS = 1000L;
+
+    private final ScheduledExecutorService scheduledExecutor =
+            Executors.newSingleThreadScheduledExecutor(
+                    r -> {
+                        Thread thread = new Thread(r, "source-reader-scheduler");
+                        thread.setDaemon(true);
+                        return thread;
+                    });
```
Comment (Hisoka-X) [inline seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java]: could you share some details about why we should introduce the `ScheduledExecutorService` in reader?

Comment (zhangshenghang) [reply seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java]: When there is no data, we should return InputStatus.NOTHIND_AVAILABLE When returning the NOTHIND_AVAILABLE state, the isAvailable() method will be executed. IsAvailable returns a future, where the longer the blocking time, the lower the Busy value.

**Comment by zhangshenghang** [comment]
@Hisoka-X @hailin0 This is a temporary solution。
a long-term solution: 
- #8358

**Review by Carl-Zhou-CN — DISMISSED** [review]
LGTM

File: seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java (hunk 1 above)
Comment (hailin0) [inline seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java]:
```suggestion
            Executors.newSingleThreadScheduledExecutor(new ThreadFactoryBuilder().setDaemon(true).setNameFormat(xxx).build());
```

**Review by hailin0 — DISMISSED** [review]
LGTM

File: seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java [hunk 2]
```diff
@@ -55,6 +59,17 @@ public class FlinkSourceReader<SplitT extends SourceSplit>
...
 
+    private volatile CompletableFuture<Void> availabilityFuture;
+
+    private static final long DEFAULT_WAIT_TIME_MILLIS = 1000L;
+
+    private final ScheduledExecutorService scheduledExecutor =
+            Executors.newSingleThreadScheduledExecutor(
+                    new ThreadFactoryBuilder()
+                            .setDaemon(true)
+                            .setNameFormat("source-reader-scheduler")
```
Comment (TyrantLucifer) [inline seatunnel-translation/seatunnel-translation-flink/seatunnel-translation-flink-common/src/main/java/org/apache/seatunnel/translation/flink/source/FlinkSourceReader.java]: name format should contain the number task

**Review by TyrantLucifer — COMMENTED** [review]
BTW, could you please offer some test results about this optimize?

**Review by Hisoka-X — DISMISSED** [review]

**Review by hailin0 — APPROVED** [review]

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**